dbengine/elasticsearch_engine_test.go

```diff
@@ -0,0 +1,143 @@
+package dbengine
+
+import (
+	composeapi "github.com/compose/gocomposeapi"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	httpmock "gopkg.in/jarcoal/httpmock.v1"
+)
+
+var _ = Describe("ElasticSearch", func() {
+	var (
+		credentials Credentials
+		dbEngine    *ElasticSearchEngine
+		deployment  composeapi.Deployment
+	)
+
+	BeforeEach(func() {
+		credentials = Credentials{
+			URI: getEnvOrDefault("ES_URL", "http://localhost:9200"),
+		}
+
+		deployment = composeapi.Deployment{
+			Connection: composeapi.ConnectionStrings{
+				Direct: []string{credentials.URI},
+			},
+		}
+	})
+
+	Context("Connection Testing", func() {
+		BeforeEach(func() {
+			dbEngine = &ElasticSearchEngine{credentials: &credentials}
+		})
+
+		It("should testConnection() successfully", func() {
+			err := dbEngine.testConnection()
+
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("should fail to testConnection() due to faulty url", func() {
+			dbEngine.credentials.URI = "http://localhost:8080"
+			err := dbEngine.testConnection()
+
+			Expect(err).Should(HaveOccurred())
```

Comment:
The unit tests are currently failing without docker. This means that the concourse build that runs the full tests are currently failing 😢 

It would be good if the "unit tests" did not require any databases running at all and the "integration tests" had all of the logic for testing with live systems to ensure all the db tests are actually run against the compose dbs.

This would mean that concourse did the full on testing and the docker dependency could be dropped from travis (which would just be running basic unit test).

I'm not 100% sure if that's the right move since the docker method _does_ make local development bit faster I assume, I guess the alternative would be to have some kind of flag to enable/disable the docker based tests.... but that way the tests would never be run against the propper compose 😕